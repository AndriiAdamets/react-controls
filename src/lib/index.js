import Button from './components/Button';
import Card from './components/Card';
import Tooltip from './components/Tooltip';
import Checkbox from './components/Checkbox';
import Input from './components/Input';
import Table from './components/Table';
export {
  Button,
  Card,
  Tooltip,
  Checkbox,
  Input,
  Table,
};