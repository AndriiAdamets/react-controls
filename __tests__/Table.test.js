import React from 'react';
import renderer from 'react-test-renderer';
import { Table } from '../src/lib';

const config = [
  {title:'First Name', accessor: 'first_name'},
  {title: 'Last Name', accessor: 'last_name', componentFn: ({item}) => (<div><b>Last Name: </b>{item.last_name}</div>)}
];

const data = [
  {first_name: 'John', last_name: 'Week'},
  {first_name: 'Anton', last_name: 'Baton'},
];

test('Table with componentFn', () => {
  const component = renderer.create(
    <Table config={config} data={data} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});