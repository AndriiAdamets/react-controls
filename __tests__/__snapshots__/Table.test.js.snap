// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table with componentFn 1`] = `
<div
  className="wrc-table"
>
  <table
    className="wrc-table__table wrc-table"
  >
    <thead
      className="wrc-table__header"
    >
      <tr>
        <th
          className="wrc-table__cell wrc-table__cell--header"
        >
          First Name
        </th>
        <th
          className="wrc-table__cell wrc-table__cell--header"
        >
          Last Name
        </th>
      </tr>
    </thead>
    <tbody
      className="wrc-table__body"
    >
      <tr
        className="wrc-table__row"
      >
        <td
          className="wrc-table__cell"
        >
          John
        </td>
        <td
          className="wrc-table__cell"
        >
          <div>
            <b>
              Last Name: 
            </b>
            Week
          </div>
        </td>
      </tr>
      <tr
        className="wrc-table__row"
      >
        <td
          className="wrc-table__cell"
        >
          Anton
        </td>
        <td
          className="wrc-table__cell"
        >
          <div>
            <b>
              Last Name: 
            </b>
            Baton
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
